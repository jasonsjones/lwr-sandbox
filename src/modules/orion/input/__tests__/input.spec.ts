import { createElement } from 'lwc';
import Element from 'orion/input';

function createInput(params = {}) {
    const element = createElement('orion-input', { is: Element }) as HTMLInputElement;
    Object.assign(element, params);
    document.body.appendChild(element);
    return element;
}
describe('orion-input', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders an input and label', () => {
        const element = createInput({ label: 'Test Input', inputId: 'test' });
        expect(element).toMatchSnapshot();
    });

    it('exposes value from html input', () => {
        const expectedText = 'orion labs';

        const element = createInput();
        const baseInput = element.shadowRoot?.querySelector('input');
        if (baseInput) {
            baseInput.value = expectedText;
            baseInput.dispatchEvent(new CustomEvent('input', { composed: true, bubbles: true }));
            expect(element.value).toBe(expectedText);
        }
    });
});
