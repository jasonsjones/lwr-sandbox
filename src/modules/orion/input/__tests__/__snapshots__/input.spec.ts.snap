// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orion-input renders an input and label 1`] = `
<orion-input>
  #shadow-root(open)
    <div
      class="relative flex flex-col"
    >
      <input
        class="peer h-10 border-b-2 border-gray-300 placeholder-transparent bg-transparent focus:border-primary-400 focus:outline-none"
        id="[shadow:guid]"
        placeholder="Test Input"
        type="text"
      />
      <label
        class="absolute -top-3.5 left-0 text-gray-600 text-sm transition-all peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-400 peer-placeholder-shown:top-2 peer-focus:-top-3.5 peer-focus:text-sm peer-focus:text-gray-600"
        for="[shadow:guid]"
      >
        Test Input
      </label>
    </div>
</orion-input>
`;
